### ISSUE TITLE
implement "b3 single" header format

### ISSUE THREAD

**Issue by adriancole** [Yelp/py_zipkin#98]
As discussed on https://github.com/openzipkin/b3-propagation/issues/21 and first implemented here: https://github.com/openzipkin/brave/blob/master/brave/src/main/java/brave/propagation/B3SingleFormat.java https://github.com/openzipkin/brave/blob/master/brave/src/test/java/brave/propagation/B3SingleFormatTest.java

Let's support at least reading "b3" header from a single string, most commonly traceid-spanid-1
It would also be nice to support optionally writing this, especially in message providers or others with constrained environments.

Brave currently has a property like this, but its name could change with feedback:
```java
    /**
     * When true, only writes a single {@link B3SingleFormat b3 header} for outbound propagation.
     *
     * <p>Use this to reduce overhead. Note: normal {@link Tracing#propagation()} is used to parse
     * incoming headers. The implementation must be able to read "b3" headers.
     */
    public Builder b3SingleFormat(boolean b3SingleFormat) {
      this.b3SingleFormat = b3SingleFormat;
      return this;
}
```

**Comment by drolando** [comment]
@adriancole How would you handle the transition phase when not all services have been updated to accept the new header?

The client cannot know if the server supports it or not since we don't do any protocol negotiation. Would people need to wait until all their servers have been upgraded before changing the client code?

**Comment by adriancole** [comment]
yes.. server first, just like when we did 128bit trace ID. that is why it
is a bit more urgent to update middleware.. in order to start that
transition.

dual propagation downstream is possible but yeah simpler to start with
making everything understand parsing both.
